File: Python/7 languages/week 1/day 01 Python/exercise 04.py
events = [
        {"name": "backup", "success": True},
        {"name": "database_sync", "success": False},
        {"name": "email_report", "success": True},
        {"name": "security_scan", "success": False}
]

def get_failed_events(events):
    failed_events = []
    for event in events:
        if not event["success"]:
            failed_events.append(event["name"])
    return failed_events

failed = get_failed_events(events)
for event in failed:
    print(f"Failed event: {event}")

File: Python/7 languages/week 1/Day 02 Typescript/exercise 04.ts
interface AutomationEvent {
    name: string;
    success: boolean;
}

const events: AutomationEvent[] = [
    { name: "backup", success: true },
    { name: "databaseSync", success: false },
    { name: "emailReport", success: true },
    { name: "securityScan", success: false }
];

function getFailedEvents(
    events: AutomationEvent[]
): string[] {
    return events.filter(event => !event.success).map(event => event.name);
}

console.log(getFailedEvents(events)); // Expected: ["databaseSync", "securityScan"]